src/mumble/API_v_1_x_x.cpp

```diff
@@ -894,6 +894,34 @@ void MumbleAPI::getChannelDescription_v_1_0_x(mumble_plugin_id_t callerID, mumbl
 	EXIT_WITH(MUMBLE_STATUS_OK);
 }
 
+void MumbleAPI::getPositionalAudioData_v_1_3_x(mumble_plugin_id_t callerID,
+											   void *positionalData,
+											   std::shared_ptr< api_promise_t > promise) {
+	if (QThread::currentThread() != thread()) {
+		// Invoke in main thread
+		QMetaObject::invokeMethod(this, "getPositionalAudioData_v_1_3_x",
+								  Qt::QueuedConnection, Q_ARG(mumble_plugin_id_t, callerID),
+								  Q_ARG(void*, positionalData),
+								  Q_ARG(std::shared_ptr< api_promise_t >, promise));
+		return;
+	}
+
+	api_promise_t::lock_guard_t guard = promise->lock();
+	if (promise->isCancelled()) {
+		return;
+	}
+
+	VERIFY_PLUGIN_ID(callerID);
+
+	auto pluginManager = Global::get().pluginManager;
+	if (!pluginManager || !pluginManager->fetchPositionalData()) {
+		EXIT_WITH(MUMBLE_EC_INTERNAL_ERROR);
+	}
+
+	positionalData = const_cast<PositionalData *>(&pluginManager->getPositionalData());
```

Comment:
Never do const casting.

Reply:
Yeah, this was just to see with a test plugin if it did indeed work (it did not), the normal function is either going to copy the object to the pointer provided (as the doc comment mentions) or if that is not deemed necessary I can change the function to a const void * although not sure the right approach in C for this kind of action